extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java

```diff
@@ -412,9 +430,14 @@ public static Stream<Path> safeWalk(Path p) {
         }
     }
 
-    private void addReflectiveClass(BuildProducer<ReflectiveClassBuildItem> reflectiveClass, boolean methods, boolean fields,
-            String... className) {
-        reflectiveClass.produce(new ReflectiveClassBuildItem(methods, fields, className));
+    private void addReflectiveClass(BuildProducer<ReflectiveHierarchyBuildItem> reflectiveClass, boolean methods,
```

Comment:
The call sites of this method would have to checked. I'm really not sure it makes sense to replace it in all cases (and might not even be necessary depending on when it's called).